### DISCUSSION TITLE
VCPKG_FORCE_DOWNLOAD_BINARIES to be able to share binarycaching artifacts between dev machine and CI

### DISCUSSION THREAD

**Discussion by meastp** [microsoft/vcpkg#18302] [Ideas]
I'm using binarycaching to share binaries between my dev machine and CI (Azure Microsoft-Hosted Agent). As I do not control the exact versions in CI, I would like to be able to force vcpkg to download and use it's own downloaded versions of CMake and PowerShell, because they contribute to the calculated abi hash for binary packages, and even a difference in patch versions disables shareing between dev machine and CI.

As a solution, I'd like a counterpart to [`VCPKG_FORCE_SYSTEM_BINARIES`](https://github.com/microsoft/vcpkg/blob/master/docs/users/config-environment.md#vcpkg_force_system_binaries), for example `VCPKG_FORCE_DOWNLOAD_BINARIES`. Bonus points for command switces to vcpkg.exe, but having the functionality in an environment variable is sufficient for my use case (the CI agent).

**Comment by JackBoosY** [comment]
vcpkg only allows downloading tools and not generated binaries. 
Is this what you want?

**Comment by meastp** [comment]
It's the tools CMake and PowerShell specifically, yes. Those tools are part of the abi hash used in binary caching.

**Comment by ras0219-msft** [comment]
Yes, we need a feature like this.

**Comment by meastp** [comment]
Isn't the feature request pretty clear? Why was this moved to discussions?

**Reply by JackBoosY** [reply]
Yes, all the vcpkg features are moving to `ideas` now.
